src/Agent.Plugins/Artifact/FileContainerProvider.cs

```diff
@@ -121,18 +121,23 @@ private async Task DownloadFileContainerAsync(Guid projectId, BuildArtifact arti
                     var directory = Path.GetDirectoryName(targetPath);
                     Directory.CreateDirectory(directory);
                     await AsyncHttpRetryHelper.InvokeVoidAsync(
-                       async () =>
-                       {
-                           using (var sourceStream = await this.DownloadFileAsync(containerIdAndRoot, projectId, containerClient, item, cancellationToken))
-                           {
-                               tracer.Info($"Downloading: {targetPath}");
-                               using (var targetStream = new FileStream(targetPath, FileMode.Create))
-                               {
-                                   await sourceStream.CopyToAsync(targetStream);
-                               }
-                           }
-                       },
-                        maxRetries: 5,
+                        async () =>
+                        {
+                            tracer.Info($"Downloading: {targetPath}");
+                            if (item.BlobMetadata != null)
```

Comment:
This is interesting, cause apparently PipelineArtifacts downloaded FCS.  Now that both tasks route here the case should work for both.

Reply:
Yes! That's the idea :)